_posts/2018-09-21-first-class-resolved-file.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: posts
+title: "First-class resolved files"
+authors:
+  - aehlig
+---
+
+In an earlier blog post, we described how a [resolved file can be used to freeze
+external dependencies](https://blog.bazel.build/2018/07/09/bazel-sync-and-resolved-file.html).
+
+- Repository rules may indicate how their arguments have to be changed
+  to reproduce identical output. This is the transition
+  from following a branch to a fixed commit. The Starlark
+  version of the `git_repository` rule already does that, and other rules
+  will follow suit soon.
+
+- With `bazel sync` there is a command to unconditionally fetch all external
+  repositories, and with the `--experimental_repository_resolved_file` option
+  all the reproducible descriptions can be collected in a file that contains
+  a Starlark value and hence is easy to work with later.
+
+In this post, we describe some recently added features.
+These changes have been comitted to the `HEAD` revision of Bazl sour and will
+be part of the 0.19 release.
+
+## Directly reading a resolved file instead of the `WORKSPACE`
+
+We made the resolved file a proper alternative to the `WORKSPACE` file. The
+corresponding option is `--experimental_resolved_file_instead_of_workspace`.
+If specified, the `WORKSPACE` file will be completely ignored, and all
+information about external repositories will be taken from the specified
+file.
+
+- In this way, the `WORKSPACE` files gets back its natural shape of describing
+  which upstream repositories a build follows. So, if following protobuf, it
+  simply reads as follows.
+
+```
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+git_repository(
```

Comment:
Why have skylib here too? I think protobuf is sufficient to get the point across

Reply:
Except that protobuf nowadays depends on skylib, and I wanted to present a working example.